Sources/TensorFlow/Core/LazyTensorShapeInference.swift

```diff
@@ -47,15 +47,30 @@ extension LazyTensorOperation {
         let status = TF_NewStatus()
         defer { TF_DeleteStatus(status) }
 
-        let inputShapes: [TensorShape] = inputs.lazy.flatMap { (input: Input) -> [TensorShape] in
+        /// Returns shape only if it has already been computed.
+        func shapeIfAvailable(_ handle: LazyTensorHandle) -> TensorShape? {
+            switch handle.handle {
+            case .symbolic(let op, let index, _):
+                if let shape = op.outputShapes[index] { return shape }
+                return nil
+            case .concrete(let tfeHandle, _): return tfeHandle.shape
+            }
+        }
+
+        LazyTensorHandle._materializationCallback("inputshape_start")
+        let inputShapes: [TensorShape?] = inputs.lazy.flatMap { (input: Input) -> [TensorShape?] in
```

Comment:
Redundant type information.
```suggestion
        let inputShapes: [TensorShape?] = inputs.lazy.flatMap { input in
```

Reply:
The type checker does not like this change. :(